2020-11-18 16:44:10.556 | INFO     | __main__:serve:85 - ��������: 192.168.0.103:64323
2020-11-18 16:44:10.558 | INFO     | __main__:serve:88 - ����ע�Ὺʼ
2020-11-18 16:44:10.562 | INFO     | __main__:serve:95 - ����ע��ɹ�
2020-11-18 16:45:31.517 | ERROR    | grpc._server:_call_behavior:435 - An error has been caught in function '_call_behavior', process 'MainProcess' (8392), thread 'ThreadPoolExecutor-0_0' (41296):
Traceback (most recent call last):

  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python38\Lib\threading.py", line 890, in _bootstrap
    self._bootstrap_inner()
    ��    �� <function Thread._bootstrap_inner at 0x0000020E58A65F70>
    �� <Thread(ThreadPoolExecutor-0_0, started daemon 41296)>

  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python38\Lib\threading.py", line 932, in _bootstrap_inner
    self.run()
    ��    �� <function Thread.run at 0x0000020E58A65CA0>
    �� <Thread(ThreadPoolExecutor-0_0, started daemon 41296)>

  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python38\Lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
    ��    ��        ��    ��        ��    �� {}
    ��    ��        ��    ��        �� <Thread(ThreadPoolExecutor-0_0, started daemon 41296)>
    ��    ��        ��    �� (<weakref at 0x0000020E5AC21450; to 'ThreadPoolExecutor' at 0x0000020E5AAA0F70>, <_queue.SimpleQueue object at 0x0000020E5AAA...
    ��    ��        �� <Thread(ThreadPoolExecutor-0_0, started daemon 41296)>
    ��    �� <function _worker at 0x0000020E5AAAAC10>
    �� <Thread(ThreadPoolExecutor-0_0, started daemon 41296)>

  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python38\Lib\concurrent\futures\thread.py", line 80, in _worker
    work_item.run()
    ��         �� <function _WorkItem.run at 0x0000020E5AAAAD30>
    �� <concurrent.futures.thread._WorkItem object at 0x0000020E5ACD57C0>

  File "C:\Users\Administrator\AppData\Local\Programs\Python\Python38\Lib\concurrent\futures\thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
             ��    ��   ��    ��       ��    �� {}
             ��    ��   ��    ��       �� <concurrent.futures.thread._WorkItem object at 0x0000020E5ACD57C0>
             ��    ��   ��    �� [<grpc._cython.cygrpc.RequestCallEvent object at 0x0000020E5ACCCF40>, <grpc._server._RPCState object at 0x0000020E5ACD56D0>, ...
             ��    ��   �� <concurrent.futures.thread._WorkItem object at 0x0000020E5ACD57C0>
             ��    �� <function _unary_response_in_pool at 0x0000020E5AAEE5E0>
             �� <concurrent.futures.thread._WorkItem object at 0x0000020E5ACD57C0>

  File "C:\Users\Administrator\Envs\mxshop_srv\lib\site-packages\grpc\_server.py", line 552, in _unary_response_in_pool
    response, proceed = _call_behavior(rpc_event, state, behavior,
                        ��              ��          ��      �� <bound method OrderServicer.CreateOrder of <userop_srv.handler.order.OrderServicer object at 0x0000020E5AAC8880>>
                        ��              ��          �� <grpc._server._RPCState object at 0x0000020E5ACD56D0>
                        ��              �� <grpc._cython.cygrpc.RequestCallEvent object at 0x0000020E5ACCCF40>
                        �� <function _call_behavior at 0x0000020E5AAEE1F0>

> File "C:\Users\Administrator\Envs\mxshop_srv\lib\site-packages\grpc\_server.py", line 435, in _call_behavior
    response_or_iterator = behavior(argument, context)
                           ��        ��         �� <grpc._server._Context object at 0x0000020E5ACD5850>
                           ��        �� userId: 1
                           ��          address: "test"
                           ��          mobile: "18787878787"
                           ��          name: "bobby"
                           ��          post: "tewtt"
                           ��          
                           �� <bound method OrderServicer.CreateOrder of <userop_srv.handler.order.OrderServicer object at 0x0000020E5AAC8880>>

  File "C:\Users\Administrator\PycharmProjects\mxshop_srvs\userop_srv\handler\order.py", line 191, in CreateOrder
    goods_srv_host, goods_srv_port = register.get_host_port(f'Service=="{settings.GOODS_SRV_NAME}"')
                                     ��        �� <function ConsulRegister.get_host_port at 0x0000020E5A20BA60>
                                     �� <common.register.consul.ConsulRegister object at 0x0000020E5ACD5C70>

  File "C:\Users\Administrator\PycharmProjects\mxshop_srvs\common\register\consul.py", line 51, in get_host_port
    return service_info["Address"], service_info["Port"]
           ��                        �� [{'ID': '1a9d9286-297a-11eb-b94e-40b0760ebfa7', 'Service': 'goods-srv', 'Tags': ['imooc', 'bobby', 'python', 'srv'], 'Meta': ...
           �� [{'ID': '1a9d9286-297a-11eb-b94e-40b0760ebfa7', 'Service': 'goods-srv', 'Tags': ['imooc', 'bobby', 'python', 'srv'], 'Meta': ...

TypeError: list indices must be integers or slices, not str
